CRM/Contribute/BAO/Contribution.php

```diff
@@ -3820,6 +3820,15 @@ public static function recordAdditionalPayment($contributionId, $trxnsData, $pay
       $participantId = CRM_Core_DAO::getFieldValue('CRM_Event_DAO_ParticipantPayment', $contributionId, 'participant_id', 'contribution_id');
     }
 
+    // fetch membership IDs registered against given contribution ID
```

Comment:
I deeply loath this function, but I think we should it in order to be as aligned with the completeOrder function as possible

loadRelatedMembershipObjects(&$ids)

- then lower down we can call the same function